clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -0,0 +1,390 @@
+    )
+    source_path_series_dicom = pd.Series(
+        find_dicom_data(imaging_data_directory), name="source_path"
+    )
+    # justifications regarding the chosen files can be found in clinica's documentation: https://github.com/aramis-lab/clinica/blob/dev/docs/Converters/UKBtoBIDS.md
+    # list of the files we want to build the bids for each modality
+    file_mod_list = [
+        "T1.nii.gz",
+        "T2_FLAIR.nii.gz",
+        "AP.nii.gz",
+        "PA.nii.gz",
+        "rfMRI.dcm",
+        "tfMRI.dcm",
+        "SWI.nii.gz",
+    ]
+
+    dataframe_nifti = pd.DataFrame.from_records(
+        source_path_series_nifti, columns=["source_zipfile", "source_filename"]
+    )
+    dataframe_dicom = pd.DataFrame.from_records(
+        source_path_series_dicom, columns=["source_zipfile", "source_filename"]
+    )
+    dataframe = pd.concat([dataframe_nifti, dataframe_dicom], 0)
+    filename = (
+        dataframe["source_filename"]
+        .apply(lambda x: Path(str(x)).name)
+        .rename("filename")
+    )
+    dataframe = dataframe[filename.isin(file_mod_list)]
+    filename = filename[filename.isin(file_mod_list)]
+    split_zipfile = dataframe["source_zipfile"].str.split("_", expand=True)
+    split_zipfile = split_zipfile.rename(
+        {0: "source_id", 1: "modality_num", 2: "source_session_number"}, axis="columns"
+    )  # .drop_duplicates()
+    split_zipfile = split_zipfile.astype({"source_id": "int64"})
+
+    df_source = pd.concat(
+        [
+            dataframe,
+            filename,
+            split_zipfile,
+        ],
+        axis=1,
+    )
+
+    return df_source
+
+
+def find_dicom_data(path_to_source_data: PathLike) -> Iterable[PathLike]:
+    from pathlib import Path
+
+    for z in Path(path_to_source_data).rglob("*.zip"):
+        if str(z).split("_")[1] == "20217":
+            yield [str(z.relative_to(path_to_source_data)), "fMRI/tfMRI.dcm"]
+        elif str(z).split("_")[1] == "20225":
+            yield [str(z.relative_to(path_to_source_data)), "fMRI/rfMRI.dcm"]
+
+
+def find_imaging_data(path_to_source_data: PathLike) -> Iterable[PathLike]:
+    from pathlib import Path
+    from zipfile import ZipFile
+
+    for z in Path(path_to_source_data).rglob("*.zip"):
+        for f in ZipFile(z).namelist():
+            if f.endswith(".nii.gz"):
+                yield [str(z.relative_to(path_to_source_data)), f]
+
+
+def intersect_data(df_source: DataFrame, df_clinical_data: DataFrame) -> DataFrame:
+    """
+    This function merges the two dataframes given as inputs based on the subject id
+    """
+
+    df_clinical = df_clinical_data.merge(
+        df_source, how="inner", right_on="source_id", left_on="eid"
+    )
+    return df_clinical
+
+
+def complete_clinical(df_clinical: DataFrame) -> DataFrame:
+    """This function uses the existing data to create the columns needed for the bids hierarchy (subject_id, ses, age_at _session, ect.)"""
```

Comment:
PEP8 complains the line is too long.

```suggestion
    """
    This function uses the existing data to create the columns needed for
    the BIDS hierarchy (subject_id, ses, age_at _session, etc...).
    """
```